src/imputors/substitute.jl

```diff
@@ -0,0 +1,90 @@
+"""
+    Substitute(; statistic=nothing)
+    Substitute(; robust=true, weights=nothing)
+
+Substitute missing values with a summary statistic over the non-missing values.
+
+# Keyword Arguments
+* `statistic`: A summary statistic function to be applied to the non-missing values.
+  This function should return a value of the same type as the input data `eltype`.
+  If this function isn't passed in then the `defaultstats` function is used to make a
+  best guess.
+* `robust`: Whether to use `median` or `mean` for continuous datasets in `defaultstats`
+* `weights`: A set of statistical weights to apply to the `mean` or `median` in `defaultstats`.
+
+# Default Rules
+Our default substitution rules defined in `defaultstats` are as follows:
+
```

Comment:
One day we should look at using SciTypes.jl for thise notion of if something is categorical or not.